src/lib/components/AssistantSettings.svelte

```diff
@@ -31,16 +34,26 @@
 
 	let files: FileList | null = null;
 	const settings = useSettingsStore();
-	let modelId =
-		assistant?.modelId ?? models.find((_model) => _model.id === $settings.activeModel)?.name;
+	let modelId = "";
 	let systemPrompt = assistant?.preprompt ?? "";
 	let dynamicPrompt = assistant?.dynamicPrompt ?? false;
+	let showModelSettings = Object.values(assistant?.generateSettings ?? {}).some((v) => !!v);
 
 	let compress: typeof readAndCompressImage | null = null;
 
 	onMount(async () => {
 		const module = await import("browser-image-resizer");
 		compress = module.readAndCompressImage;
+
+		if (assistant) {
+			modelId = assistant.modelId;
+		} else {
+			if (models.map((model) => model.id).includes($settings.activeModel)) {
+				modelId = $settings.activeModel;
+			} else {
+				modelId = models[0].id;
+			}
```

Comment:
```suggestion
			modelId = models.find((model) => model.id === $settings.activeModel)?.id ?? models[0].id;
```

didn't run it, maybe we can just use this nullish coalescing

Reply:
tested it, it works, thanks!